### PR TITLE
Handle when fragmentation settings are undefined

### LINKED ISSUE #38: Decoding JSON fails with unexpected string in response
#### JSON Property Returns "undefined" when `int` is Expected

An error is logged when trying to decode a JSON response from `/pools/default`. It is likely that the call to json.NewDecoder(...).Decode(model) is failing when attempting to decode the "undefined" string in the JSON property `percentage`.

**This failure appears to stop allowing reporting to occur for the node.**

[`err = json.NewDecoder(response.Body).Decode(model)`](https://github.com/newrelic/nri-couchbase/blob/c721f68e7324873d24b6dbfee4e2ae5341e72679/src/client/client.go#L89)

##### Log Message

```text
time="2020-01-31T22:41:30-08:00" level=error msg="Integration command failed" error="exit status 1" instance= integration=com.newrelic.couchbase prefix=config/couchbase stderr="[ERR] Could not complete collection: received an unexpected response from endpoint '/pools/default': json: cannot unmarshal string into Go struct field ViewFragmentationThreshold.percentage of type int\n" working-dir=/var/db/newrelic-infra/newrelic-integrations
```

##### JSON Property

The JSON property returns the value of "undefined" where an integer is expected.

```json
"viewFragmentationThreshold":{"percentage":"undefined","size":"undefined"}
```

##### [Couchbase Cluster-wide Settings](https://docs.couchbase.com/server/current/rest-api/rest-autocompact-get.html#cluster-wide-response)

This appears to happen when calling `/settings/autoCompaction`. However, it would be good to ensure that any values returned as "undefined", a valid JSON type, for numerics can be handled accordingly.

```json
{
    "autoCompactionSettings": {
        "databaseFragmentationThreshold": {
            "percentage": undefined,
            "size": "undefined"
        },
        "parallelDBAndViewCompaction": false,
        "viewFragmentationThreshold": {
            "percentage": undefined,
            "size": "undefined"
        }
    },
    "purgeInterval": 3
}
```

##### OS

- [x] Ubuntu, versions 12.04, 14.04, and 16.04 (LTS  versions)

### PR DESCRIPTION
**Pull request by camdencheek**
#### Description of the changes

Fixes #38

### CODE REVIEW

File: src/entities/collector.go [hunk 1]
```diff
@@ -166,3 +168,26 @@ func GetBucketCollectors(args *arguments.ArgumentList, i *integration.Integratio
 
 	return collectors, nil
 }
+
+func sanitizeAutocompactionSettings(response *definition.PoolsDefaultResponse) {
+	switch response.AutoCompactionSettings.DatabaseFragmentationThreshold.Percentage.(type) {
+	case int:
+		break
+	default:
+		response.AutoCompactionSettings.DatabaseFragmentationThreshold.Percentage = nil
```
Comment (icpenguins) [inline src/entities/collector.go]: nitpick: would it be possible to add debug logging here that the value was converted?

File: src/entities/collector.go [hunk 2]
```diff
@@ -166,3 +168,26 @@ func GetBucketCollectors(args *arguments.ArgumentList, i *integration.Integratio
...
+	}
+
+	switch response.AutoCompactionSettings.IndexFragmentationThreshold.Percentage.(type) {
+	case int:
+		break
+	default:
+		response.AutoCompactionSettings.IndexFragmentationThreshold.Percentage = nil
```
Comment (icpenguins) [inline src/entities/collector.go]: nitpick: would it be possible to add debug logging here that the value was converted?

File: src/entities/collector.go [hunk 3]
```diff
@@ -166,3 +168,26 @@ func GetBucketCollectors(args *arguments.ArgumentList, i *integration.Integratio
...
+	}
+
+	switch response.AutoCompactionSettings.ViewFragmentationThreshold.Percentage.(type) {
+	case int:
+		break
+	default:
+		response.AutoCompactionSettings.ViewFragmentationThreshold.Percentage = nil
```
Comment (icpenguins) [inline src/entities/collector.go]: nitpick: would it be possible to add debug logging here that the value was converted?

File: src/entities/cluster_test.go [hunk 4]
```diff
@@ -57,3 +57,37 @@ func createClusterCollector(i *integration.Integration, httpClient *http.Client,
...
 }
+
+func Test_sanitizeAutocompactionSettings(t *testing.T) {
+	input := &definition.PoolsDefaultResponse{
+		AutoCompactionSettings: &definition.AutoCompactionSettings{
+			DatabaseFragmentationThreshold: &definition.DatabaseFragmentationThreshold{
+				Percentage: 46,
+			},
+			IndexFragmentationThreshold: &definition.IndexFragmentationThreshold{
+				Percentage: "undefined",
```
Comment (mariomac) [inline src/entities/cluster_test.go]: I don't think this patch fixes the `undefined` issue as it seem to come from the limitations in the Go JSON unmarshaller, as the undefined is not received as a JSON string but as a Javascript symbol:

```json
{
   "autoCompactionSettings": {
       "databaseFragmentationThreshold": {
           "percentage": undefined,
           "size": "undefined"
       },
       "parallelDBAndViewCompaction": false,
       "viewFragmentationThreshold": {
           "percentage": undefined,
           "size": "undefined"
       }
   },
   "purgeInterval": 3
}
```

**Pull request merged**